js/views/modals/listingDetail/Listing.js

```diff
@@ -168,13 +282,30 @@ export default class extends BaseModal {
         shipsFreeToMe: this.shipsFreeToMe,
         ownListing: this.model.isOwnListing,
         displayCurrency: app.settings.get('localCurrency'),
+        shipsFromCountry: app.settings.get('country'),
       }));
 
       super.render();
 
       this._$deleteListing = null;
       this._$shipsFreeBanner = null;
       this._$popInMessages = null;
+      this._$photoSection = null;
+
+      // commented out until variants are available
+      // this.$('.js-variantSelect').select2();
+
+      this.$photoSelected = this.$('.js-photoSelected');
+      this.$photoSelected.backgroundDraggable();
+
+      const shippingDest = this.$('#shippingDestinations');
+
+      shippingDest.select2({
+        data: this.select2CountryData,
+        placeholder: app.polyglot.t('listingDetail.shipToPlaceholder'),
+      });
+
+      shippingDest.val(app.settings.get('country')).trigger('change');
```

Comment:
I'm a little confused.

> To elaborate, what we really want here is a way to set the selected element when select2 is initialized

Well, that has nothing to do with synthetically firing the 'change' event which is what I'm flagging as a poor practice.

I'm just suggesting changing this:

```
shippingDest.val(app.settings.get('country')).trigger('change');
```

to

```
shippingDest.val(app.settings.get('country'));
setShippingDestination(app.settings.get('country'));
```

That also requires the `setShippingDestination` signature to accept the data it needs rather than a jQuery event. An `onSetShippingDestination` handler should accept the jQuery event and then call `setShippingDestination`.

==========

Anyways, `shippingDest.val(app.settings.get('country'))` part doesn't exactly set the default value on initialization, but it does set it immediately afterwards, which should accomplish the same thing.

If you really want to set it on initialization, look at how I build the select2 markup in the EditListing and Settings modals. Essentially, I don't pass any data into the select2 element at initialization. I just build the <option> elements in the template and set the default one by giving it a "selected" attribute. Then, when I do initialize select2, it builds it's data and default selection from the markup.